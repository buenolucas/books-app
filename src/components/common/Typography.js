import React from 'react';
import PropTypes from 'prop-types';
import {getFontStyleObject} from '../../utils/font';
import {StyleSheet} from 'react-native';
import styled from 'styled-components/native';
import Theme from './../../Theme';

const Text = styled.Text`
  ${getFontStyleObject()}
  color: ${props => props.theme.colors.text};
`;

const Typography = props => {
  const {children, style, type} = props;
  !Theme.typography[type] &&
    console.warn(`Typography: There is no ${type} type in typography.`);
  const textStyles = [Theme.typography[type], style];
  return (
    <Text {...props} style={textStyles}>
      {children}
    </Text>
  );
};
Typography.propTypes = {
  type: PropTypes.string,
  children: PropTypes.any,
  style: PropTypes.any,
};

Typography.defaultProps = {
  type: 'body',
};

export default Typography;
