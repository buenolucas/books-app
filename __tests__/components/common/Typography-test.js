import React from 'react';
import {shallow} from 'enzyme';
import Typography from '../../../src/components/common/Typography';

describe('Typography', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<Typography type="body">test text</Typography>);
      expect(component).toMatchSnapshot();
    });
  });
});
