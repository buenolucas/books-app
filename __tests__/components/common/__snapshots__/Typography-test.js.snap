// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Rendering should match to snapshot 1`] = `
<Styled(Text)
  style={
    Array [
      Object {
        "fontFamily": "Roboto",
        "fontSize": 14,
        "fontStyle": "normal",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
  type="body"
>
  test text
</Styled(Text)>
`;
